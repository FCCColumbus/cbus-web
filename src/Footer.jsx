
import links from './links'

function Footer() {

    return (

        <footer>
            <nav>
                <ul>
                    {links.map((link, i) => i === 0 ? '' : <li key={link.text}><a href={link.href}>{link.text === 'Home' ? <img src='/fcc-logo.svg' alt='fcc logo' className='header-logo' /> : link.text}</a></li>)}
                </ul>
            </nav>
            <p>Copyright Free Code Camp Columbus 2023 - {new Date().getFullYear()}</p>

        </footer>

    );

}

export default Footer