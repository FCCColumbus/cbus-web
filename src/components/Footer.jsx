import links from '../links'

function Footer() {
  return (
    <footer>
      <nav>
        <ul>
          {links.map((link, i) =>
            i === 0
              ? ''
              : (
                <li key={link.text}>
                  <a href={link.href}>
                    {
                      link.text === 'Home'
                        ? <img src='/fcc-logo.svg' alt='fcc logo' className='header-logo' />
                        : link.text
                    }
                  </a>
                </li>
              )
          )}
        </ul>
      </nav>
      <p>Copyright Free Code Camp Columbus 2023 - {new Date().getFullYear()}</p>
      <a href="https://www.netlify.com" target="_blank" rel="noreferrer" className="image-wrapper" data-testid="image-wrapper"> <img src="https://www.netlify.com/v3/img/components/netlify-color-accent.svg" alt="Deploys by Netlify" /> </a>
    </footer>
  );
}

export default Footer;
