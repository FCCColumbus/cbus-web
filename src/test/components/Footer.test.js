import renderer from 'react-test-renderer';

import Footer from '../../components/Footer';
import { render, screen } from '@testing-library/react';

describe('Footer', () => {
  it('renders', () => {
    const view = renderer.create(<Footer />);
    expect(view).toMatchSnapshot();
  });

  it('displays "Discord" as the third item', () => {
    render(<Footer />);
    const thirdItem = screen.getByText('Discord');
    expect(thirdItem).toBeInTheDocument();
  });
});
