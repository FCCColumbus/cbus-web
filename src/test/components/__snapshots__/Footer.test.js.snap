// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<footer>
  <nav>
    <ul>
      <li>
        <a
          href="#about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="https://www.freecodecamp.org/"
        >
          FreeCodeCamp
        </a>
      </li>
      <li>
        <a
          href="https://discord.gg/aF8skEe9nq"
        >
          Discord
        </a>
      </li>
      <li>
        <a
          href="https://github.com/FCCColumbus"
        >
          GitHub
        </a>
      </li>
      <li>
        <a
          href="#events"
        >
          Events
        </a>
      </li>
      <li>
        <a
          href="https://github.com/FCCColumbus/cbus-web/graphs/contributors"
        >
          Contributors
        </a>
      </li>
    </ul>
  </nav>
  <p>
    Copyright Free Code Camp Columbus 2023 - 
    2023
  </p>
</footer>
`;
